driver-core/src/main/com/mongodb/connection/netty/NettyStreamFactoryFactory.java

```diff
@@ -34,28 +36,108 @@
 public class NettyStreamFactoryFactory implements StreamFactoryFactory {
 
     private final EventLoopGroup eventLoopGroup;
+    private final Class<? extends SocketChannel> socketChannelClass;
     private final ByteBufAllocator allocator;
 
     /**
      * Construct an instance with the given {@code EventLoopGroup} and {@code ByteBufAllocator}.
      *
      * @param eventLoopGroup the non-null event loop group
      * @param allocator the non-null byte buf allocator
+     * @deprecated Use {@link NettyStreamFactoryFactory#builder()} instead to construct the {@code  NettyStreamFactoryFactory}.
      */
+    @Deprecated
     public NettyStreamFactoryFactory(final EventLoopGroup eventLoopGroup, final ByteBufAllocator allocator) {
-        this.eventLoopGroup = notNull("eventLoopGroup", eventLoopGroup);
-        this.allocator = notNull("allocator", allocator);
+        this(builder().eventLoopGroup(eventLoopGroup).allocator(allocator));
     }
 
     /**
-     * Construct an instance with the default {@code EventLoopGroup} and {@code ByteBufAllocator}.
+     * Gets a builder for an instance of {@code NettyStreamFactoryFactory}.
+     * @return the builder
+     * @since 3.3
      */
-    public NettyStreamFactoryFactory() {
```

Comment:
The no-args constructor has to remain as well as a deprecated constructor.  It breaks binary compatibility to remove it.